### PR TITLE
[docs] Trash

### PR DESCRIPTION
**Pull request by alexyarosh**
Rename archive to Trash, pull it out of History, and add explanations of how trashing affects related items.

Not sure if it should have a backport label considering RC1 was cut

Things to think about:

Name and content:
- This goes into “Organization” topic so I thought “”the means/goal of organization action” (delete and restore) would be a better name than ”the place where it goes” (Trash).
But then this also made me think that people might come to that article to learn about deleting items that don’t go into trash, so I added a bunch of links to other delete-like actions.

Article organization:

- There are a lot of repeated instructions which I don’t like. On the other hand, I wanted to have detailed section titles for better searcheability. Alternative would be to have delete/restore instructions as one section, and then separate the effects of delete actions into sections.

Downstream effects:
- I think I noted most of the possible consequences/interactions of deleting something, but let me know if I missed something!

### CODE REVIEW

File: docs/README.md
Comment (jeff-bruemmer) [inline docs/README.md:129]:
```suggestion
- [Trash](./exploration-and-organization/delete-and-restore.md)
```

I get that you're going for keywords `delete` and `restore`, but we can use trash here, as that's what the page covers.

File: docs/exploration-and-organization/delete-and-restore.md
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:11]: You'll need to note here that you can search for items in the trash, if you need to.

File: docs/exploration-and-organization/delete-and-restore.md
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:115]: The text is confusing. maybe a table? Columns are: Model in collection, Trash, permanently deleted.

So something like: 

**Item** |**In trash**|**Permanently deleted**|**Restored**
-----|-----|-----|-----
Question based on model|Works as normal|Breaks|Works as normal
Dashboard that includes model|Model card is removed from the dashboard| Model card is removed from the dashboard |Card is restored to the dashboard

**Review by jeff-bruemmer — CHANGES_REQUESTED** [review]
- We should prioritize human eyeballs first, then machine eyeballs. To that end, we could significantly consolidate these docs. I'm not categorically against duplication, but here the repetition makes it easy to see the trees, but not the forest.
- The forest: we should talk about how to delete and restore items in general, then list the effects of a trash move for each type of item. I doubt many people will ever search for how to delete something. What people might want to read is info about what happens to related items when they delete something. E.g., "what happens to items and sub-collections when you move a collection to the trash", or dashboard cards/subscriptions, or models with downstream questions, etc.
- In general, use existing docs and expand them, rather than repeat elsewhere. E.g., the search section should be a one-liner that links to the [search section](https://github.com/metabase/metabase/blob/master/docs/exploration-and-organization/exploration.md#advanced-search) (this linking will eliminate at least one screenshot).
- Speaking of screenshots: I prefer a little more visual context in the images. Right now the menus are isolated/disconnected from the larger UI so they're difficult to situate in context. And since the menus are straightforward, it's more important to show people where to find the menus, not how to use them.
- I know our table support isn't stellar, but there are some opportunities for them here to enumerate item cases (I gave one hasty example in-line re: model trashing).

I know I'm asking for a lot of changes, but this is an excellent first draft, @alexyarosh. Very thorough. Made it easy to edit and identify the important stuff we should highlight.

**Comment by alexyarosh** [comment]
Thanks for awesome feedback @jeff-bruemmer ! Check out the new draft

File: docs/exploration-and-organization/delete-and-restore.md
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:33]:
```suggestion
To move an item (question, dashboard, model, or collection) to the Trash:
```

File: docs/exploration-and-organization/delete-and-restore.md
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:59]: Maybe just a intro sentence for each table for more context? Something like:  "What happens to related items when you delete a question?"

File: docs/exploration-and-organization/delete-and-restore.md
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:35]: Think we need to to also talk about trashing items from the collection view, in particular bulk trashing.

File: docs/exploration-and-organization/delete-and-restore.md
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:89]:
```suggestion
Moving an item to Trash doesn't delete the item completely: you'll be able to restore the item from the Trash.
```
Slight clarification.

File: docs/exploration-and-organization/delete-and-restore.md
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:96]: Maybe some warning about how you won't be able to recover the item. It's lost to the void.

File: docs/exploration-and-organization/delete-and-restore.md
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:108]:
```suggestion
You won't see archived SQL snippets in Trash. To see archived SQL Snippets, you need to [access them from the Snippet menu](../questions/native-editor/sql-snippets.md#snippet-menu).
```

**Review by jeff-bruemmer — APPROVED** [review]
Left a few comments to address before merging.

And remember to remove unused images from the PR.

Nice work.

**Comment by alexyarosh** [comment]
Trash is not making it to v50 so putting this on hold for now

File: docs/exploration-and-organization/delete-and-restore.md
Comment (appleby) [inline docs/exploration-and-organization/delete-and-restore.md:126]: Hi @alexyarosh, this link is failing the `yarn run docs-lint-links` stage for me in CI.

Unclear to me what the link target should be. Did you forget to `git add` the `segments-and-metrics.md` file, or should the link maybe point to `segments.md` instead?

Comment (appleby) [reply docs/exploration-and-organization/delete-and-restore.md:126]: Link to the CI failure for reference

https://github.com/metabase/metabase/actions/runs/11227506788/job/31245666789

Comment (alexyarosh) [reply docs/exploration-and-organization/delete-and-restore.md:126]: @appleby done #48478

Comment (appleby) [reply docs/exploration-and-organization/delete-and-restore.md:126]: Ah makes, sense. Thanks!

**Pull request merged**
